geopandas/io/parquet.py

```diff
@@ -215,88 +254,64 @@ def _to_parquet(df, path, compression="snappy", index=None, **kwargs):
+def _to_feather(df, path, index=None, **kwargs):
     """
-    Load a Parquet object from the file path, returning a GeoDataFrame.
+    Write a GeoDataFrame to the Feather format.
 
-    You can read a subset of columns in the file using the ``columns`` parameter.
-    However, the structure of the returned GeoDataFrame will depend on which
-    columns you read:
+    Any geometry columns present are serialized to WKB format in the file.
 
-    * if no geometry columns are read, this will raise a ``ValueError`` - you
-      should use the pandas `read_parquet` method instead.
-    * if the primary geometry column saved to this file is not included in
-      columns, the first available geometry column will be set as the geometry
-      column of the returned GeoDataFrame.
+    Requires 'pyarrow' >= 0.17.
 
-    Requires 'pyarrow'.
+    WARNING: this is an initial implementation of Feather file support and
+    associated metadata.  This is tracking version 0.1.0 of the metadata
+    specification at:
+    https://github.com/geopandas/geo-arrow-spec
+
+    This metadata specification does not yet make stability promises.  As such,
+    we do not yet recommend using this in a production setting unless you are
+    able to rewrite your Feather files.
 
     .. versionadded:: 0.8
 
     Parameters
     ----------
     path : str, path object
-    columns : list-like of strings, default=None
-        If not None, only these columns will be read from the file.  If
-        the primary geometry column is not included, the first secondary
-        geometry read from the file will be set as the geometry column
-        of the returned GeoDataFrame.  If no geometry columns are present,
-        a ``ValueError`` will be raised.
-    **kwargs
-        Any additional kwargs passed to pyarrow.parquet.read_table().
-
-    Returns
-    -------
-    GeoDataFrame
+    index : bool, default None
+        If ``True``, always include the dataframe's index(es) as columns
+        in the file output.
+        If ``False``, the index(es) will not be written to the file.
+        If ``None``, the index(ex) will be included as columns in the file
+        output except `RangeIndex` which is stored as metadata only.
+    compression : {'snappy', 'gzip', 'brotli', None}, default 'snappy'
+        Name of the compression to use. Use ``None`` for no compression.
+    kwargs
+        Additional keyword arguments passed to pyarrow.feather.write_feather().
     """
-
-    import_optional_dependency(
-        "pyarrow", extra="pyarrow is required for Parquet support."
+    feather = import_optional_dependency(
+        "pyarrow.feather", extra="pyarrow is required for Feather support."
     )
-    from pyarrow import parquet
+    # TODO move this into `import_optional_dependency`
+    import pyarrow
 
-    kwargs["use_pandas_metadata"] = True
-    table = parquet.read_table(path, columns=columns, **kwargs)
+    if pyarrow.__version__ < "0.17":
```

Comment:
Sorry - this was my fault.  I opted to go simplistic in porting `import_optional_dependency` from `pandas`; we now need version support there.  Crossref: https://github.com/pandas-dev/pandas/blob/master/pandas/compat/_optional.py#L47-L111

Reply:
No, no, I actually asked you to keep it simple ;) We should only port what we needed, and in the previoius PR, this was not yet needed. 

Now, the version matching can also not be copied from pandas, as it works a bit differently there (also, we are importing a submodule which has not a version itself .. which makes it a bit more complicated)